keras/backend/tensorflow_backend.py

```diff
@@ -2897,9 +2897,22 @@ def rnn(step_function, inputs, initial_states,
     if mask is not None:
         if mask.dtype != tf.bool:
             mask = tf.cast(mask, tf.bool)
-        if len(mask.get_shape()) == ndim - 1:
-            mask = expand_dims(mask)
-        mask = tf.transpose(mask, axes)
```

Comment:
This shows that we're on the "wrong track", as the masking shape should not depend on input (which `axes` are derived from)